### PR TITLE
Add a note to clarify limitations of arithmetic opcodes

### PR DESCRIPTION
**Pull request by A60AB5450353F40E**
Copied from: https://gitlab.com/bitcoin-cash-node/bitcoin-cash-node/-/blob/master/src/script/script.h#L213

### CODE REVIEW

File: protocol/blockchain/script.md [hunk 1]
```diff
@@ -125,6 +125,8 @@ Op codes marked with **(do not use)** are disallowed and will make a transaction
 
 ### Arithmetic
 
+Numeric opcodes (OP_1ADD, etc) are restricted to operating on 4-byte integers. The semantics are subtle, though: operands must be in the range [-2^31 +1...2^31 -1], but results may overflow (and are valid as long as they are not used in a subsequent numeric operation). CScriptNum enforces those semantics by storing results as an int64 and allowing out-of-range values to be returned as a vector of bytes but throwing an exception if arithmetic is done or the result is interpreted as an integer.
```
Comment (emergent-reasons) [inline protocol/blockchain/script.md]: The last part of this looks language specific and probably should not be included.

`numeric operation` could also probably more precisely be stated with opcodes again. Together, you might use text like this:

> Numeric opcodes (OP_1ADD, etc) are restricted to operating on 4-byte integers. The semantics are subtle, though: operands must be in the range [-2^31 +1...2^31 -1], but results may overflow (and are valid as long as they are not subsequently used with numeric opcodes).

**Comment by A60AB5450353F40E** [comment]
updated, meta q: is git commit --amend followed by force push the right way?

**Comment by dagurval** [comment]
Thanks for the submission :-)

> (and are valid as long as they are not used in a subsequent numeric operation).

From reading the Script interpreter code, it looks like trying to use the result will cause a `scriptnum_error` exception to be thrown, and be caught by a generic `catch (...)` exception handler that sets the error code to `ScriptError::UNKNOWN`, failing the script execution.

Perhaps we can clarify and say that script execution will fail if used in subsequent numeric operation?

(I'm happy to merge as is though, this is a clear improvement)

_Edit: I also checked the interpreter implementation in BU. BU returns a helpful `SCRIPT_ERR_NUMBER_OVERFLOW` instead of `UNKNOWN`_

File: protocol/blockchain/script.md (hunk 1 above)
Comment (dagurval) [reply protocol/blockchain/script.md]: I can see that the text was updated with your feedback, resolving thread

**Comment by dagurval** [comment]
Merging, thanks!

**Pull request merged**